tests/crashes/120421.rs

```diff
@@ -1,12 +0,0 @@
-//@ known-bug: #120421
-//@ compile-flags: -Zlint-mir
-
-#![feature(never_patterns)]
-
-enum Void {}
-
-fn main() {
-    let res_void: Result<bool, Void> = Ok(true);
-
-    for (Ok(mut _x) | Err(!)) in [res_void] {}
-}
```

Comment:
I deleted this because it felt like an entirely artificial construct and I couldn't find a sensible name to keep it under. What surprises me is that it took something so complicated to break the feature back when it was explicitly not lowering to MIR correctly. It's probably not minimal